apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ClientPool.cs

```diff
@@ -87,50 +92,137 @@ public override int GetHashCode()
             }
         }
 
-        private class ClientPoolEntry
+        private class CredentialClientPool
         {
-            private readonly ClientPoolKey _key;
-            private SpannerClient _client;
-            private volatile Task<SpannerClient> _creationTask;
+            private readonly ClientCredentialKey _key;
+            private SortedList<SpannerClientCreator, SpannerClientCreator> _multiClientPool = 
+                new SortedList<SpannerClientCreator, SpannerClientCreator>();
```

Comment:
imagine it suffers the same as sortedlist -- it doesnt allow comparable to be zero (so different instance, same priority).